tests/scripts/thread-cert/network_diag.py

```diff
@@ -0,0 +1,335 @@
+
+import common
+import ipaddress
+import mle
+
+
+class TlvType(IntEnum):
+    EXT_ADDRESS = 0
+    ADDRESS16 = 1
+    MODE = 2
+    POLLING_PERIOD = 3
+    CONNECTIVITY = 4
+    ROUTE64 = 5
+    LEADER_DATA = 6
+    NETWORK_DATA = 7
+    IPV6_ADDRESS_LIST = 8
+    MAC_COUNTERS = 9
+    BATTERY_LEVEL = 14
+    SUPPLY_VOLTAGE = 15
+    CHILD_TABLE = 16
+    CHANNEL_PAGES = 17
+    TYPE_LIST = 18
+    MAX_CHILD_TIMEOUT = 19
+
+
+class Ipv6AddressList:
+
+    def __init__(self, addresses: [ipaddress.IPv6Address]):
+        self._addresses = addresses
+
+    @property
+    def addresses(self):
+        return self._addresses
+
+    def __eq__(self, other):
+        common.expect_the_same_class(self, other)
+        return self.addresses == other.addresses
+
+    def __repr__(self):
+        return f'Ipv6AddressList({self.addresses})'
+
+
+class Ipv6AddressListFactory:
+
+    def parse(self, data, message_info):
+        addresses = []
+        while data.tell() < message_info.length:
+            addresses.append(ipaddress.IPv6Address(data.read(16)))
+        return Ipv6AddressList(addresses)
+
+
+class MacCounters:
+
+    def __init__(self, counters: [int]):
+        self._counters = counters
+
+    @property
+    def if_in_unknown_protos(self):
+        return self._counters[0]
+
+    @property
+    def if_in_errors(self):
+        return self._counters[1]
+
+    @property
+    def if_out_errors(self):
+        return self._counters[2]
+
+    @property
+    def if_in_ucast_pkts(self):
+        return self._counters[3]
+
+    @property
+    def if_in_broadcast_pkts(self):
+        return self._counters[4]
+
+    @property
+    def if_in_discards(self):
+        return self._counters[5]
+
+    @property
+    def if_out_ucast_pkts(self):
+        return self._counters[6]
+
+    @property
+    def if_out_broadcast_pkts(self):
+        return self._counters[7]
+
+    @property
+    def if_out_discards(self):
+        return self._counters[8]
+
+    @property
+    def counters(self):
+        return self._counters
+
+    def __eq__(self, other):
+        common.expect_the_same_class(self, other)
+
+        return self.counters == other.counters
+
+    def __repr__(self):
+        return ('MacCounters(' +
+                f'if_in_unknown_protos={self.if_in_unknown_protos}, ' +
+                f'if_in_errors={self.if_in_errors}, ' +
+                f'if_out_errors={self.if_out_errors}, ' +
+                f'if_in_ucast_pkts={self.if_in_ucast_pkts}, ' +
+                f'if_in_broadcast_pkts={self.if_in_broadcast_pkts}, ' +
+                f'if_in_discards={self.if_in_discards}, ' +
+                f'if_out_ucast_pkts={self.if_out_ucast_pkts}, ' +
+                f'if_out_broadcast_pkts={self.if_out_broadcast_pkts}, ' +
+                f'if_out_discards={self.if_out_discards})')
+
+
+class MacCountersFactory:
+
+    def parse(self, data, message_info):
+        return MacCounters(struct.unpack('>9I', data.read(4 * 9)))
+
+
+class BatteryLevel:
```

Comment:
why not just use `int`?

Reply:
Many other single field TLVs defines a class for it. So I'd like to keep the same approach.